gluten-core/src/main/scala/org/apache/spark/sql/execution/ColumnarCollapseTransformStages.scala

```diff
@@ -159,6 +165,9 @@ case class ColumnarCollapseTransformStages(
 
   private def insertWholeStageTransformer(plan: SparkPlan): SparkPlan = {
     plan match {
+      case t: CartesianProductExecTransformer =>
+        WholeStageTransformer(insertInputIteratorTransformer(t))(
```

Comment:
@zhouyuan To create CartesianRDD, it is necessary to get resultant RDD from left and right. I have added a new `WholeStageTransformer` whenever there is CartesianProduct so that a single RDD is returned from both side. This is similar to default Spark behavior.

This is also similar to BroadcastHashJoin where broadcast needs to be executed first to perform the join. To achieve this AQE adds "BroadcastQueryStage" and it is not transformed into Gluten which automatically adds new WholeStageTransformer for it in Gluten.